src/lib/server/tmpl_tokenize.c

```diff
@@ -1790,7 +1800,7 @@ static inline int tmpl_attr_afrom_attr_substr(TALLOC_CTX *ctx, tmpl_attr_error_t
 	 *	Don't alter the fr_strerror buffer, may contain useful
 	 *	errors from the dictionary code.
 	 */
-	if (!at_rules->allow_unresolved) {
+	if (!at_rules->allow_unresolved && !(at_rules->allow_wildcard && fr_sbuff_is_char(name, '['))) {
```

Comment:
It this more automatic wildcard so if an index isn't provided we return all the instances? Or what's meant to be going on here.

Reply:
IIRC this is a sanity check for `&request[*]`.  There's only one `request`, so adding filters on it should be forbidden

Reply:
If that's the case then check if it's a list attribute instead of passing rules down would be better?

Reply:
Unless you can think of another situation where we need to allow/deny filters?

Reply:
It allows `&request.[*]` when `allow_unresolved` is false.  The first parsing pass will have extracted `&request`, then we are left with a blank attr ref which won't be found in the dictionaries.  Testing for `allow_wildcard` and the next character being `[` allows this to fall through to call `tmpl_attr_ref_afrom_unresolved_substr` which will add the undefined attr ref and parse the filter.